wasm2c/wasm-rt-wasi.c

```diff
@@ -0,0 +1,798 @@
+// This file contains modified versions of the emscripten runtime (as well as the wasi support it includes ) adapted for
```

Comment:
This should probably not be called `wasi` because it implements a lot more than that.   Maybe `wasm-rt-syscalls.c` or `wasm-rt-externs.c` or maybe we should create a new sub-directory so we can split this into maybe file?